packages/babel-preset-env/src/normalize-options.js

```diff
@@ -15,6 +15,48 @@ const validIncludesAndExcludes = new Set([
   ...defaultWebIncludes,
 ]);
 
+const validRegExp = (regexp: string): boolean => {
+  try {
+    new RegExp(regexp);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+const selectPlugins = (regexp: RegExp): Array<string> =>
+  Array.from(validIncludesAndExcludes).filter(item => item.match(regexp));
+
+const pluginToRegExp = (plugin: any): RegExp => {
+  return plugin instanceof RegExp
+    ? plugin
+    : new RegExp(normalizePluginName(plugin));
+};
+
+const populatePlugins = (
+  pluginList: Array<RegExp>,
+  regexp: RegExp,
+): Array<string> => pluginList.concat(selectPlugins(regexp));
+
+const isValidPlugin = (plugin: any): boolean =>
+  validRegExp(plugin) && selectPlugins(pluginToRegExp(plugin)).length > 0;
+
+const expandIncludesAndExcludes = (
+  plugins: Array<string>,
+  type: string,
+): Array<string> => {
+  const invalidRegExp = plugins.filter(plugin => !isValidPlugin(plugin));
+
+  invariant(
+    invalidRegExp.length === 0,
+    `Invalid RegExp: The given regular expression '${invalidRegExp.join(
+      ", ",
+    )}' passed to the '${type}' option are not valid.`,
+  );
+
+  return plugins.map(pluginToRegExp).reduce(populatePlugins, []);
```

Comment:
And if the value is a `string` and it's a valid regExp and could be converted with `pluginToRegExp` it's still a `string`, so there will be the 3-rd `new RegExp(plugin)` call.